examples/Huggingface_Transformers/Pipeline_handler_generalized.py

```diff
@@ -0,0 +1,172 @@
+            '''
+            self.map_to_preproc_fn = {
+                "image-classification":self.preprocess_image_classification_single,
+                "sentiment-analysis":self.preprocess_sentiment_analysis_single,
+                "text-classification":self.preprocess_sentiment_analysis_single
+            }
+            pass
+        '''
+        Takes a list of raw inference requests received via POST requests, preprocesses and returns the output as a list
+        '''
+        def __call__(self, data):
+            if self.map_to_preproc_fn[task] is not None:
+                return map(self.map_to_preproc_fn[task] ,data)
+            else:
+                return data
+        def preprocess_image_classification_single(self, file):
+            image = file.get("data") or file.get("body")
+            if isinstance(image, str):
+                # if the image is a string of bytesarray.
+                image = base64.b64decode(image)
+            # If the image is sent as bytesarray
+            if isinstance(image, (bytearray, bytes)):
+                image = Image.open(io.BytesIO(image))
+            return image
+
+        #Function to read .txt file and convert it into a string
+        def preprocess_sentiment_analysis_single(self, file):
+            text = file.get("data") or file.get("body")
+            # Decode text if not a str but bytes or bytearray
+            if isinstance(text, (bytes, bytearray)):
+                text = text.decode("utf-8", errors='ignore')
+            return text
+
+    class HFPipelinePostprocessDispatcher:
+        '''
+            `map_to_postproc_fn` returns a function for the task that can postprocess a single inference output to convert it to serializeable form with proper JSON formatting,
+            else if no postprocessing is required, `map_to_postproc_fn` returns None
+        '''
+        def __init__(self):
+            self.map_to_postproc_fn = {
+                "image-classification":None,
+                "sentiment-analysis":None,
+                "text-classification":None
+            }
+            pass 
+        def __call__(self, data):
+            if self.map_to_postproc_fn[task] is not None:
+                return map(self.map_to_postproc_fn[task], data)
+            else:
+                return data
+    
+    def __init__(self):
+        super().__init__()
+        self.tokenizer = None
+        self.preprocess_dispatcher = self.HFPipelinePreprocessDispatcher()
+        self.postprocess_dispatcher = self.HFPipelinePostprocessDispatcher()
+
+
+    def load_model(self, device_id, model_name, hf_models_folder = "/home/model-server/HF-models"):
+        logger.info('Entered `load_model` function')
+        model_folder = os.path.join(hf_models_folder, model_name)
+
+        logger.info("Creating pipeline")
+        pipe = pipeline(task=task, framework=framework, model=model_folder, device = device_id)
```

Comment:
@msaroufim while exploring I stumbled upon #1579 and from what I understand `model_parallel` is a parameter to allocate an LLM instance like GPT-2 to multiple GPUs instead of one GPU?

If I understand it right, this can be done with any model supported by HF `pipeline` as they all support [`device_map`](https://huggingface.co/docs/transformers/main_classes/pipelines#transformers.pipeline.device_map), but they warn that only one of `device` & `device_map` attribute must be used.  

Thought this might be of interest to you.